tools/windows/install-help/cal/FinalizeInstall.cpp

```diff
@@ -1,5 +1,68 @@
-#include "TargetMachine.h"
 #include "stdafx.h"
+#include "PropertyReplacer.h"
+#include "TargetMachine.h"
+#include <fstream>
+
+bool updateYamlConfig(CustomActionData &customActionData)
+{
+    std::wstring inputConfig;
+
+    // Read config in memory. The config should be small enough
+    // and we control its source - so it's fine to allocate up front.
+    {
+        std::wifstream inputConfigStream(datadogyamlfile);
+
+        inputConfigStream.seekg(0, std::ios::end);
+        size_t fileSize = inputConfigStream.tellg();
+        if (fileSize <= 0)
+        {
+            WcaLog(LOGMSG_STANDARD, "ERROR: datadog.yaml file empty !");
+            return false;
+        }
+        inputConfig.reserve(fileSize);
+        inputConfigStream.seekg(0, std::ios::beg);
+
+        inputConfig.assign(std::istreambuf_iterator<wchar_t>(inputConfigStream), std::istreambuf_iterator<wchar_t>());
+    }
+
+    std::wregex re(L"^[ #]*api_key:(.*)");
+    std::match_results<std::wstring::const_iterator> results;
+    if (std::regex_search(inputConfig, results, re))
+    {
+        auto api_key = results[1].str();
+        api_key.erase(api_key.begin(),
+                      std::find_if(api_key.begin(), api_key.end(), [](int ch) { return !std::isspace(ch); }));
+        if (api_key.length() > 0)
+        {
+            WcaLog(LOGMSG_STANDARD, "API key already present in configuration - not modifying it");
+            return true;
+        }
+    }
+
+    std::vector<std::wstring> failedToReplace;
+    inputConfig =
+        replace_yaml_properties(inputConfig, [&customActionData](std::wstring const &propertyName) -> std::optional<std::wstring> {
+            std::wstring propertyValue;
+            if (customActionData.value(propertyName, propertyValue))
+            {
+                return propertyValue;
+            }
+            return std::nullopt;
+        },
+        &failedToReplace);
+
```

Comment:
For troubleshooting installation problems, we should log what properties are supplied and what their values are (except certain properties that should
be redacted like API_KEY and DDAGENTUSER_PASSWORD, in which case we should still log ****** to indicate they were passed in and accepted.  (maybe here or maybe where they're parsed)